File: back_repo/src/main/java/com/d101/frientree/service/LeafService.java
package com.d101.frientree.service;

import com.d101.frientree.dto.leaf.request.LeafGenerationRequest;
import com.d101.frientree.dto.leaf.response.LeafComplaintResponse;
import com.d101.frientree.dto.leaf.response.LeafConfirmationResponse;
import com.d101.frientree.dto.leaf.response.LeafGenerationResponse;
import com.d101.frientree.dto.leaf.response.LeafViewResponse;
import org.springframework.http.ResponseEntity;

public interface LeafService {

    // category 선택해서 조회하기
    ResponseEntity<LeafConfirmationResponse> confirm(String leafCategory);

    ResponseEntity<LeafGenerationResponse> generate(LeafGenerationRequest leafGenerationRequest);

    ResponseEntity<LeafComplaintResponse> complain(Long leafId);

    ResponseEntity<LeafViewResponse> view(Long leafId);

}

File: android_repo/data/src/main/java/com/d101/data/api/LeafService.kt
package com.d101.data.api

import com.d101.data.model.ApiResponse
import com.d101.data.model.ApiResult
import com.d101.data.model.leaf.request.LeafCreationRequest
import com.d101.data.model.leaf.response.LeafResponse
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Path

interface LeafService {
    @POST("/leaf")
    suspend fun sendLeaf(
        @Body leafCreationRequest: LeafCreationRequest,
    ): ApiResult<ApiResponse<Boolean>>

    @POST("/leaf/{leafId}")
    suspend fun reportLeaf(
        @Path("leafId") leafId: Long,
    ): ApiResult<ApiResponse<Boolean>>

    @GET("/leaf/{category}")
    suspend fun getLeaf(
        @Path("category") leafCategory: Int,
    ): ApiResult<ApiResponse<LeafResponse>>

    @GET("/leaf/view")
    suspend fun getLeafViews(): ApiResult<ApiResponse<Int>>
}
